pep-9999.rst

```diff
@@ -0,0 +1,861 @@
+---------------------------------------------------------------------------------------------
+
+Rather than using the beta period for rolling releases, another option would be
+to alternate between traditional stable releases (for 3.8.x, 3.10.x, etc), and
+release series that used the new rolling release cadence (for 3.9.x, 3.11.x,
+etc).
+
+This idea suffers from the same core problem as PEP 598 and PEP 602: it imposes
+changes on end users that are happy with the status quo without offering them
+any clear compensating benefit.
+
+It's also affected by one of the main concerns raised against PEP 598: at least
+some core developers and end users strongly prefer that no particular semantics
+be assigned to the *value* of any of the numbers in a release version. These
+community members instead prefer that all the semantic significance be
+associated with the *position* within the release number that is changing.
+
+By contrast, the rolling beta release proposal aims to address that concern by
+ensuring that the proposed changes in policy all revolve around whether a
+particular release is an alpha release, beta release, release candidate, or
+final release.
+
+
+Why not use Calendar Versioning for the rolling release stream?
+---------------------------------------------------------------
+
+Steve Dower's initial write-up of this proposal [1_] suggested the use of
+calendar versioning for the rolling release stream (so the first rolling
+pre-release after Python 3.8.0 would have been Python 2019.12 rather than
+3.9.0b1).
+
+Paul Moore pointed out [2_] two major practical problems with that proposal:
+
+* it isn't going to be clear to users of the calendar-based versions where they
+  stand in relation to the traditionally numbered versions
+* it breaks ``Python-Requires`` metadata processing in packaging tools with
+  no clear way of fixing it reliably (since all calendar versions would appear
+  as newer than any standard version)
+
+This PEP aims to address both of those problems by using the established beta
+version numbers for the rolling releases.
+
+As an example, consider the following question: "Does Python 2021.12 include
+all the new features released in Python 3.9.0?". With calendar versioning on
+the rolling releases, that's impossible to answer without consulting a release
+calender to see when 3.9.0rc1 was branched off from the rolling release series.
+
+By constrast, the equivalent question for rolling beta releases is
+straightforward to answer: "Does Python 3.10.0b2 include all the new features
+released in Python 3.9.0?". Just from formulating the question, the answer is
+clearly "Yes, unless they were provisional features that got removed".
+
+The beta numbering approach also avoids other questions raised by the calendar
+versioning concept, such as how ``sys.version_info``, ``PY_VERSION_HEX``,
+``site-packages`` directory naming, and installed Python binary and extension
```

Comment:
The list entry here is genuinely "``site-packages`` directory naming", since the parent directory gets the version number in it on non-Windows systems.

Reply:
Ah, I see. Thanks for the clarification.